src/mscorlib/src/System/Decimal.cs

```diff
@@ -378,7 +378,7 @@ public int CompareTo(Object value)
             if (value == null)
                 return 1;
             if (!(value is Decimal))
-                throw new ArgumentException(Environment.GetResourceString("Arg_MustBeDecimal"));
+                throw new ArgumentException(Environment.GetResourceString("Arg_MustBeDecimal"), nameof(value));
```

Comment:
There are many similar `CompareTo` methods on primitive types. This improvement should be either applied to all of them or none of them. @tarekgh @AlexGhiondea Do you think it is a good idea to apply similar fix to all `CompareTo` methods?